domains/certificates/Issuer/Issuer.Worker/GranularCertificateIssuer/EnergyMeasuredEventHandler.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.Numerics;
+using System.Threading.Tasks;
+using CertificateEvents;
+using CertificateEvents.Primitives;
+using Issuer.Worker.MasterDataService;
+
+namespace Issuer.Worker.GranularCertificateIssuer;
+
+public class EnergyMeasuredEventHandler : IEnergyMeasuredEventHandler
+{
+    private readonly IMasterDataService masterDataService;
+
+    public EnergyMeasuredEventHandler(IMasterDataService masterDataService) => this.masterDataService = masterDataService;
+
+    public async Task<ProductionCertificateCreated?> Handle(EnergyMeasured @event)
+    {
+        var masterData = await masterDataService.GetMasterData(@event.GSRN);
+        if (masterData is null)
```

Comment:
This can be seen as a SLAP-violation (SLAP: Single Level of Abstraction Principle) in the `Handle` method, but since the method is short I found it okay at this point. 
But I do not have a strong opinion on that.  But maybe you have that ;) ?

Reply:
Who wants to be the one with all the "strong" opinions? :)
But if I see plastic in the nature i would pick it up if it didnt require too much work